go-routesapi/workload/router.go

```diff
@@ -8,16 +8,20 @@ import (
 	"github.com/signadot/routesapi/go-routesapi/watched"
 )
 
-type Router interface {
-	GetTarget(containerPort int32, rks ...string) string
-	GetTargetWithContext(ctx context.Context, containerPort int32, rks ...string) (string, error)
-}
-
 type router struct {
 	watched watched.BaselineWatched
 }
 
 func NewRouter(cfg *Config) (Router, error) {
+	var cliInfo *routesapi.ClientInfo
+	if cfg.ClientInfo != nil {
+		cliInfo = cfg.ClientInfo
+	} else {
+		// by default support virtual workloads
```

Comment:
I think this might confuse folks using the library for selective consumption, because it can break.

Would you be ok with defaulting to not enabling virtual and enabling it specifically in the devmesh?